tony-core/src/main/java/com/linkedin/tony/TaskExecutor.java

```diff
@@ -118,15 +123,28 @@ public static void main(String[] args) {
         }
 
         // Execute the user command
-        HashMap<String, String> extraEnv = new HashMap<String, String>(executor.shellEnv) {
-          {
-            put(Constants.TB_PORT, String.valueOf(executor.tbPort));
-            put(Constants.PY4JGATEWAY, String.valueOf(executor.gatewayServerPort));
-            put(Constants.JOB_NAME, String.valueOf(executor.jobName));
-            put(Constants.TASK_INDEX, String.valueOf(executor.taskIndex));
-            put(Constants.CLUSTER_SPEC, String.valueOf(executor.clusterSpec));
+        HashMap<String, String> extraEnv = new HashMap<>(executor.shellEnv);
+        extraEnv.put(Constants.TB_PORT, String.valueOf(executor.tbPort));
+        extraEnv.put(Constants.PY4JGATEWAY, String.valueOf(executor.gatewayServerPort));
+        extraEnv.put(Constants.JOB_NAME, String.valueOf(executor.jobName));
+        extraEnv.put(Constants.TASK_INDEX, String.valueOf(executor.taskIndex));
+        extraEnv.put(Constants.CLUSTER_SPEC, String.valueOf(executor.clusterSpec));
+
+        // Reset taskCommand if we are running in Docker mode.
+        if (executor.isDockerMode) {
+          assert(executor.dockerImagePath != null);
+          Utils.executeShell(executor.dockerBinaryPath + " pull " + executor.dockerImagePath,
+                             10, extraEnv);
+          executor.taskCommand = executor.dockerBinaryPath
+                                 + " run"
+                                 + " -p" + String.valueOf(executor.tbPort) + ":" + String.valueOf(executor.tbPort)
+                                 + " -p" + String.valueOf(executor.gatewayServerPort) + ":" + String.valueOf(executor.gatewayServerPort);
```

Comment:
Can you add a comment here explaining what

```
-p tbPort:tbPort -p gwServerPort:gwServerPort
```

does for those unfamiliar with Docker?